moped-editor/src/views/staff/StaffForm.js

```diff
@@ -0,0 +1,296 @@
+import React from "react";
+import { useUserApi } from "./helpers";
+import { useForm, Controller } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+import { gql, useQuery } from "@apollo/react-hooks";
+import {
+  Button,
+  FormControl,
+  FormControlLabel,
+  FormLabel,
+  FormHelperText,
+  Grid,
+  InputLabel,
+  TextField,
+  makeStyles,
+  MenuItem,
+  Radio,
+  RadioGroup,
+  Select,
+} from "@material-ui/core";
+
+const WORKGROUPS_QUERY = gql`
+  query GetWorkgroups {
+    moped_workgroup {
+      workgroup_id
+      workgroup_name
+    }
+  }
+`;
+
+const useStyles = makeStyles(() => ({
+  formSelect: {
+    minWidth: 195,
+  },
+}));
+
+const initialFormValues = {
+  first_name: "",
+  last_name: "",
+  title: "",
+  email: "",
+  password: "",
+  workgroup: "",
+  workgroup_id: "",
+  roles: "moped-viewer",
+  status_id: "1",
+};
+
+const roles = [
+  { value: "moped-viewer", name: "Viewer" },
+  { value: "moped-editor", name: "Editor" },
+  { value: "moped-admin", name: "Admin" },
+];
+
+const statuses = [
+  { value: "1", name: "Active" },
+  { value: "0", name: "Inactive" },
+];
+
+const staffValidationSchema = yup.object().shape({
+  first_name: yup.string().required(),
+  last_name: yup.string().required(),
+  title: yup.string().required(),
+  workgroup: yup.string().required(),
+  workgroup_id: yup.string().required(),
+  email: yup.string().required(),
+  password: yup.string().required(),
+  roles: yup.string().required(),
+  status_id: yup.string().required(),
+});
```

Comment:
Very cool way to validate data! :)

Reply:
Thanks, Yup plugs right into React Hook Form and makes it nice to add. 👍 Also, great catch on the validation helper text for the workgroup field. I added that in the next branch with the edit/delete features. 🙏